src/Field.php

```diff
@@ -508,6 +509,36 @@ public function useAlias(): bool
 
     // }}}
 
+    // {{{ Scope condition
+
+    /**
+     * Returns arguments to be used for query on this field based on the condition.
+     *
+     * @param string|null $operator one of Scope\Condition operators
+     * @param mixed       $value    the condition value to be handled
+     *
+     * @todo: value is array
+     */
+    public function getQueryArguments($operator, $value): array
+    {
+        if ($persistence = $this->owner->persistence) {
```

Comment:
if persistence is not set, we should probably throw, is there anything against?

Reply:
Is it mandatory to have the persistence?
In case of no persistence we simply do not cast the value and return 'as is'

Reply:
when we prepare something to query, I do not see a reason why to prepare/handle usecase when we will need the persistence for quering a few ops after

Reply:
Yes, I would say the best is to expect `$this->model->persistence` is here and directly use it.
`checkPersistence` is more for checking if a specific method exists in the persistence which we do not need here

Reply:
> Yes, I would say the best is to expect `$this->model->persistence` is here and directly use it.

yes, I think the same